src/main/java/org/opensearch/knn/index/engine/EngineResolver.java

```diff
@@ -30,33 +35,39 @@ public KNNEngine resolveEngine(
         KNNMethodContext knnMethodContext,
         boolean requiresTraining
     ) {
+        KNNEngine knnEngine = KNNEngine.FAISS;
```

Comment:
Lets keep return statements as is and just add a function before return that logs if deprecated. Im worried that we might be altering resolution logic with this

Reply:
I wanted to avoid that pattern, but sure. Let me work that through.